src/androidTest/kotlin/com/criptext/mail/scenes/emaildetail/data/ReadEmailsWorkerTest.kt

```diff
@@ -0,0 +1,136 @@
+package com.criptext.mail.scenes.emaildetail.data
+
+import android.support.test.rule.ActivityTestRule
+import android.support.test.runner.AndroidJUnit4
+import com.criptext.mail.Config
+import com.criptext.mail.androidtest.TestActivity
+import com.criptext.mail.androidtest.TestDatabase
+import com.criptext.mail.api.HttpClient
+import com.criptext.mail.db.DeliveryTypes
+import com.criptext.mail.db.models.*
+import com.criptext.mail.scenes.emaildetail.workers.ReadEmailsWorker
+import com.criptext.mail.utils.DateUtils
+import com.criptext.mail.utils.MockedResponse
+import com.criptext.mail.utils.enqueueResponses
+import io.mockk.mockk
+import okhttp3.mockwebserver.MockWebServer
+import org.amshove.kluent.shouldEqualTo
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.sql.Timestamp
+
+@RunWith(AndroidJUnit4::class)
+class ReadEmailsWorkerTest {
+
+    @get:Rule
+    val mActivityRule = ActivityTestRule(TestActivity::class.java)
+
+    private val mockedThreadId = Timestamp(System.currentTimeMillis()).toString()
+    private lateinit var db: TestDatabase
+    private lateinit var mockWebServer: MockWebServer
+    private val activeAccount = ActiveAccount(name = "Tester", recipientId = "tester",
+            deviceId = 1, jwt = "__JWTOKEN__", signature = "")
+
+    private lateinit var httpClient: HttpClient
+
+    @Before
+    fun setup() {
+        db = TestDatabase.getInstance(mActivityRule.activity)
+        db.resetDao().deleteAllData(1)
+        mockWebServer = MockWebServer()
+        mockWebServer.start()
+
+        httpClient = HttpClient.Default(authScheme = HttpClient.AuthScheme.jwt,
+                baseUrl = mockWebServer.url("/mock").toString(), connectionTimeout = 7000L,
+                readTimeout = 7000L)
+    }
+
+    @After
+    fun teardown() {
+        if (Config.mockCriptextHTTPRequests)
+            mockWebServer.close()
+    }
+
+    private fun newWorker(emailIds: List<Long>, metadataKeys: List<Long>): ReadEmailsWorker =
+            ReadEmailsWorker(dao = db.emailDao(), activeAccount = activeAccount,
+                    httpClient = httpClient, publishFn = {}, emailIds = emailIds,
+                    metadataKeys = metadataKeys)
+
+
+    @Test
+    fun should_send_request_to_read_any_unread_emails_and_return_the_number_of_updated_emails() {
+
+        if (Config.mockCriptextHTTPRequests) {
```

Comment:
There are very few tests that we want to have the flexibility of choosing whether to run it with a mock server or the real thing. this isn't one of them because it needs a session on the server, so this check is not needed.